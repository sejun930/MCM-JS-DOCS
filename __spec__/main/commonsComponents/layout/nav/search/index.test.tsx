import { MutableRefObject, useRef } from "react";

import { render } from "@testing-library/react";
import NavSearchPage from "src/main/commonsComponents/layout/nav/search";

jest.mock("react", () => ({
  ...jest.requireActual("react"),
  useRef: jest.fn(),
}));

export default describe("Layout Nav Search Page", () => {
  const _inputRef = useRef() as MutableRefObject<HTMLInputElement>;

  // 스냅샷
  test("Snapshot", () => {
    const { container } = render(
      <NavSearchPage
        _inputRef={_inputRef}
        changeSearch={() => {}}
        search="test"
      />
    );
    expect(container).toMatchSnapshot();
  });

  // 검색을 위한 input 태그 존재 여부 검증
  test("Check input tag", () => {
    const { container } = render(
      <NavSearchPage
        _inputRef={_inputRef}
        changeSearch={() => {}}
        search="test"
      />
    );

    // wrapper 존재 여부 검증
    const wrapperEle = container.querySelector(".nav-search-wrapper");
    expect(wrapperEle).toBeInTheDocument();

    // input 존재 여부 검증
    const inputEle = wrapperEle?.querySelector(".nav-search-input");
    expect(inputEle).toBeInTheDocument();
  });
});
