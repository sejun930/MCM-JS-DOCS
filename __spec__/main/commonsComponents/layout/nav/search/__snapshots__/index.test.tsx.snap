// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Nav Search Page Snapshot 1`] = `
<div>
  <div
    class="nav-search-wrapper css-1pry9ao"
  >
    xw
    <fieldset
      class="css-1opjr3t"
    >
      <legend>
        모듈 검색
      </legend>
      <div
        class="css-1rbly99"
      >
        <div
          class="css-2gjeij"
        >
          <input
            class="nav-search-input css-1wxelb4"
            maxlength="20"
            placeholder="모듈 검색"
            type="text"
          />
          <div
            class="css-iju4mq"
          >
            <button
              class="mcm-close-button-unit css-1yn8fww"
            />
          </div>
        </div>
      </div>
    </fieldset>
  </div>
</div>
`;
